tests/core/filtering/test_contract_data_filters.py

```
@@ -151,48 +144,51 @@ def test_data_filters_with_fixed_arguments(
+@pytest.mark.parametrize('api_style', ('v4', 'build_filter'))
 @given(vals=array_values())
+@settings(max_examples=5, deadline=None)
 def test_data_filters_with_list_arguments(
         web3,
         emitter,
-        Emitter,
         wait_for_transaction,
-        emitter_event_ids,
         call_as_instance,
         create_filter,
-        vals):
+        api_style,
+        vals,
+        request):
 
     matching, non_matching = vals
 
-    if call_as_instance:
-        event_filter = create_filter(emitter, [
-            'LogListArgs', {
-                'filter': {"arg1": matching}}])
+    if api_style == 'build_filter':
+        filter_builder = emitter.events.LogListArgs.build_filter()
+        filter_builder.args['arg1'].match_single(matching)
+        event_filter = filter_builder.deploy(web3)
+
+        txn_hashes = []
+        txn_hashes.append(emitter.functions.logListArgs(
+            arg0=matching,
+            arg1=matching).transact({'gasPrice': 1, 'gas': 100000, 'nonce': 0}))
+        txn_hashes.append(emitter.functions.logListArgs(
+            arg0=non_matching,
+            arg1=non_matching).transact({'gasPrice': 1, 'gas': 100000, 'nonce': 1}))
+        txn_hashes.append(emitter.functions.logListArgs(
+            arg0=non_matching,
+            arg1=matching).transact({'gasPrice': 1, 'gas': 100000, 'nonce': 2}))
+        txn_hashes.append(emitter.functions.logListArgs(
+            arg0=matching,
+            arg1=non_matching).transact({'gasPrice': 1, 'gas': 100000, 'nonce': 3}))
+
+        for txn_hash in txn_hashes:
+            wait_for_transaction(web3, txn_hash)
+
+        log_entries = event_filter.get_new_entries()
+        assert len(log_entries) == 2
+        assert log_entries[0]['transactionHash'] == txn_hashes[0]
+        assert log_entries[1]['transactionHash'] == txn_hashes[2]
     else:
-        event_filter = create_filter(Emitter, [
-            'LogListArgs', {
-                'filter': {"arg1": matching}}])
```

So this is # 2 from: https://github.com/ethereum/web3.py/pull/953#issuecomment-416095677 right? Can you remind me *what* was failing silently (since this was a working test before).

---

There was an issue with the way the regex would be built when array arguments were included.  The regex sized each item in the encoded data to be 32 bytes.

---

the test was marked XFAIL before.

---

Perfect, thanks.